### ISSUE TITLE
Add logits test on CUDA + FP16 for BetterTransformer

### ISSUE THREAD

**Issue by fxmarty** [huggingface/optimum#1256]
### Feature request

This is currently not tested - only calls to `generate` are, and on tiny models it is not meaningful. We should rather tests the output logits like done for fp32.

### Motivation

Catch errors

### Your contribution

/
